src/app/(proper_react)/redesign/(authenticated)/user/dashboard/View.tsx

```diff
@@ -47,6 +47,7 @@ export type Props = {
   userScanData: LatestOnerepScanData;
   isEligibleForFreeScan: boolean;
   isEligibleForPremium: boolean;
+  isPremiumUser: boolean;
```

Comment:
Instead of adding yet another parameter, can't we just derive this from `props.user`?

```suggestion
```

Reply:
Oh, so I'm used to have a clear separation of controller / view where the view is purely presentation and as simple as possible in terms of not containing any logic or dependency. It doesn't seem that we abide by that principle too much. I'm ok blurring the line here if we don't have that type of convention

Reply:
Oh yeah we've definitely not been doing MVC, though I see how the name "View" is confusing :sweat_smile: The distinction here is more server-side logic vs. client-side logic - it's fine to have logic in the view, as long as it's related to what to display to the user. We could discuss further splitting up the views into [container and presentational components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0) in the tech meeting, if you're interested - but see the disclaimer at the top of that article :)